File: framework/apis/PetstoreSingleGetApi.ts
import { APIResponse } from '@playwright/test';
import { BaseAPI } from '@core/BaseAPI';

/**
 * Generated API client — Petstore single GET
 * @generated WebPilot
 */
export class PetstoreSingleGetApi {
  constructor(private readonly client: BaseAPI) {}

  async getApibaseurlpetfindbystatusstatusavaila(): Promise<APIResponse> {
    const response = await this.client.get('{{apiBaseUrl}}/pet/findByStatus?status=available');
    await this.client.assertStatus(response, 200);
    return response;
  }
}

File: framework/apis/petstore_single_get_api.py
from playwright.sync_api import APIResponse

from framework.core.base_api import BaseAPI


class PetstoreSingleGetApi:
    def __init__(self, client: BaseAPI) -> None:
        self.client = client

    def find_available_pets(self) -> APIResponse:
        response = self.client.get(
            "https://petstore.swagger.io/v2/pet/findByStatus?status=available"
        )
        self.client.assert_status(response, 200)
        return response
